pkg/gen/awaitComments.go

```diff
@@ -35,21 +66,25 @@ const DeploymentAwaitComment = preamble + `
    'Progressing', whose 'status' member is set to 'True', and whose 'reason' is
    'NewReplicaSetAvailable'. For generation <= 1, this status field does not exist,
    because it doesn't do a rollout (i.e., it simply creates the Deployment and
-   corresponding ReplicaSet), and therefore there is no rollout to mark as 'Progressing'.`
-
-const StatefulSetAwaitComment = preamble + `
-1. The value of 'spec.replicas' matches '.status.replicas', '.status.currentReplicas',
-   and '.status.readyReplicas'.
-2. The value of '.status.updateRevision' matches '.status.currentRevision'.`
-
-const PodAwaitComment = preamble + `
+   corresponding ReplicaSet), and therefore there is no rollout to mark as 'Progressing'.
+`
+	case kinds.Ingress:
+		comment += `
+1.  Ingress object exists.
+2.  Endpoint objects exist with matching names for each Ingress path (except when Service
+    type is ExternalName).
+3.  Ingress entry exists for '.status.loadBalancer.ingress'.
+`
+	case kinds.Pod:
+		comment += `
 1. The Pod is scheduled ("PodScheduled"" '.status.condition' is true).
 2. The Pod is initialized ("Initialized" '.status.condition' is true).
 3. The Pod is ready ("Ready" '.status.condition' is true) and the '.status.phase' is
    set to "Running".
-Or (for Jobs): The Pod succeeded ('.status.phase' set to "Succeeded").`
-
-const ServiceAwaitComment = preamble + `
+Or (for Jobs): The Pod succeeded ('.status.phase' set to "Succeeded").
+`
+	case kinds.Service:
+		comment += `
 1. Service object exists.
 2. Related Endpoint objects are created. Each time we get an update, wait ~5-10 seconds
```

Comment:
Is there a reason we have a range of 5-10 seconds here versus a fixed amount?

Reply:
Looks like it actually has a 10s timer set. Updated the doc.